### PR TITLE
[Model] MixHop for Node Classification task

### PR DESCRIPTION
**Pull request by xnuohz**
## Description

This DGL example implements the GNN model proposed in the paper [MixHop: Higher-Order Graph Convolution Architectures via Sparsified Neighborhood Mixing](https://arxiv.org/abs/1905.00067).

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/mixhop/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# DGL Implementation of MixHop
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:1]: "DGL Implementation of MixHop" -> "DGL implementations of MixHop" for the consistency across the file.

File: examples/pytorch/mixhop/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+This DGL example implements the GNN model proposed in the paper [MixHop: Higher-Order Graph Convolution Architectures via Sparsified Neighborhood Mixing](https://arxiv.org/abs/1905.00067). For the original implementation, see [here](https://github.com/samihaija/mixhop).
+
+## Example implementor
+
+This example was implemented by [xnuohz](https://github.com/xnuohz) during his intern at the AWS Shanghai AI Lab.
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:7]: I think you can simply replace this section with something like below

```
Contributor: [Name](github link)
```

File: examples/pytorch/mixhop/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+dgl 0.5.2
+numpy 1.19.4
+pandas 1.1.4
+tqdm 4.53.0
+torch 1.7.0
+```
+
+### The graph datasets used in this example
+
+The DGL's built-in Cora, Pubmed and Citeseer GraphDataset. Dataset summary:
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:22]: "GraphDataset" -> "datasets"

File: examples/pytorch/mixhop/README.md (hunk 1 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:1]: Done.

File: examples/pytorch/mixhop/README.md (hunk 2 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:7]: Done.

File: examples/pytorch/mixhop/README.md (hunk 3 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:22]: Done.

File: examples/pytorch/mixhop/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+pandas 1.1.4
+tqdm 4.53.0
+torch 1.7.0
+```
+
+### The graph datasets used in this example
+
+The DGL's built-in Cora, Pubmed and Citeseer datasets. Dataset summary:
+
+| Dataset | Nodes | Edges | Feats | Classes | # Train Nodes | # Val Nodes | # Test Nodes |
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:22]: Also add "#" before "Nodes", "Edges", "Feats", "Classes"

File: examples/pytorch/mixhop/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+--early-stopping   int     Early stopping rounds.              Default is 200.
+--lr               float   Adam optimizer learning rate.       Default is 0.5.
+--lamb             float   L2 regularization coefficient.      Default is 0.0005.
+--step-size        int     Period of learning rate decay.      Default is 40.
+--gamma            float   Factor of learning rate decay.      Default is 0.01.
+--hid-dim          int     Hidden layer dimensionalities.      Default is 60.
+--num-layers       int     Number of GNN layers.               Default is 4.
+--input-dropout    float   Dropout applied at input layer.     Default is 0.7.
+--layer-dropout    float   Dropout applied at hidden layers.   Default is 0.9.
+--p                list    Powers list of adjacency matrix.    Default is [0, 1, 2].
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:57]: "Powers list" -> "List of powers"

File: examples/pytorch/mixhop/README.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+```
+
+###### Examples
+
+The following commands learn a neural network and predict on the test set.
+Training a MixHop model on the default dataset.
+```bash
+$ python main.py
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:65]: Can you remove "$ "?

File: examples/pytorch/mixhop/README.md (hunk 4 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:22]: Done.

File: examples/pytorch/mixhop/README.md (hunk 5 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:57]: Done.

File: examples/pytorch/mixhop/README.md (hunk 6 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:65]: Removed.

File: examples/pytorch/mixhop/main.py [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    np.random.seed(seed)
+    random.seed(seed)
+    torch.backends.cudnn.deterministic = True
+
+def main(args):
+    # Step 0: set random seed
+    if args.seed >= 0:
+        setup_seed(args.seed)
```
Comment (mufeili) [inline examples/pytorch/mixhop/main.py:168]: Let's remove the seeding part. Otherwise, 100 runs make no sense.

Comment (xnuohz) [reply examples/pytorch/mixhop/main.py]: Done.

File: examples/pytorch/mixhop/README.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+# Pubmed:
+python main.py --gpu 0 --dataset Pubmed --lr 0.5 --input-dropout 0.7 --lamb 5e-3 --hid-dim 60 --num-layers 3
+```
+
+### Performance
+
+| Dataset | Cora | Pubmed | Citeseer |
+| :-: | :-: | :-: | :-: |
+| Accuracy(original paper) | 0.818 | 0.800 | 0.714 |
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:90]: There are 5 entries in two tables in the original paper. Explain which entry you are referring to.

File: examples/pytorch/mixhop/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+| Accuracy(TF) | 0.610(0.156) | 0.746(0.065) | 0.700(0.017) |
```
Comment (mufeili) [inline examples/pytorch/mixhop/README.md:91]: I will simply replace "TF" with "official code"

File: examples/pytorch/mixhop/main.py [hunk 10]
```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+# -*- coding:utf-8 -*-
```
Comment (mufeili) [inline examples/pytorch/mixhop/main.py:2]: You don't need these two lines as DGL only supports Python 3.

**Review by mufeili — APPROVED** [review]
Overall a great job! This PR is good once you address the minor comments.

File: examples/pytorch/mixhop/main.py (hunk 10 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/main.py:2]: Done.

File: examples/pytorch/mixhop/README.md (hunk 8 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:90]: Done.

File: examples/pytorch/mixhop/README.md (hunk 9 above)
Comment (xnuohz) [reply examples/pytorch/mixhop/README.md:91]: Done.
